hazelcast-client/src/main/java/com/hazelcast/client/config/ClientFlakeIdGeneratorConfig.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.util.Preconditions;
+
+/**
+ * The {@code ClientFlakeIdGeneratorConfig} contains the configuration for the client
+ * regarding {@link com.hazelcast.core.HazelcastInstance#getFlakeIdGenerator(String)
+ * Flake ID Generator}.
+ */
+public class ClientFlakeIdGeneratorConfig {
+
+    /**
+     * Default value for {@link #getPrefetchCount()}.
+     */
+    public static final int DEFAULT_PREFETCH_COUNT = 100;
+
+    /**
+     * Default value for {@link #getPrefetchValidity()}.
+     */
+    public static final long DEFAULT_PREFETCH_VALIDITY = 10000;
+
+    private String name;
+    private int prefetchCount = DEFAULT_PREFETCH_COUNT;
+    private long prefetchValidity = DEFAULT_PREFETCH_VALIDITY;
```

Comment:
I think we should rename this field to `prefetchValidityInMillis`. It should specify the time unit

Reply:
I renamed to `prefetchValidityMillis`, this seemed to me more consistent with other settings.